extensions/attention-stream/src/youtube/utils/traverseFishing.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Recrusive tree traversal of JSON object:
+ * Returns or deletes one or multiple keys
+ * depending on parameters
+ */
+const traverse = (
+  jsonObject,
+  shouldDelete,
+  returnFirst,
+  keys,
+  keepKeys = false
+) => {
+  let results = [];
+
+  const traverseHelper = (jsonObject) => {
+    for (var key in jsonObject) {
+      if (keys.includes(key)) {
+        if (shouldDelete) {
+          delete jsonObject[key];
+        } else {
+          if (keepKeys) {
+            results.push([key, jsonObject[key]]);
+          } else {
+            results.push(jsonObject[key]);
+          }
+        }
+
+        if (returnFirst) {
+          break;
+        }
+      }
+      if (jsonObject[key] !== null && typeof jsonObject[key] == "object") {
+        // Recurse
+        traverseHelper(jsonObject[key]);
+      }
+    }
+  };
+
+  traverseHelper(jsonObject);
+  return results;
+};
+
+const fishFor = (key, jsonObject) => {
```

Comment:
Why `fish`? Is this to imply that there may or may not be a result? I think it's OK just unusual :)

Reply:
Do you mean the variable name? I was sort of having a bit of fun :p but also yes "get" felt wrong since the intention is to sometimes use this to check if the key exists. Open to other name suggestions though!